libvast/src/system/index.cpp

```diff
@@ -869,6 +869,8 @@ index(index_actor::stateful_pointer<index_state> self,
       }
       self->state.inmem_partitions.drop(partition_id);
       self->state.persisted_partitions.erase(partition_id);
+      self->request(self->state.meta_index, caf::infinite, atom::erase_v,
+                    partition_id);
```

Comment:
Is the effect of `request` without `then` different from `send`?

Reply:
Only slightly. It allows for adding a `.then` handler that matches either `()` or `(caf::error)`. Internally the only difference is the additional hidden request ID parameter. We should always use `request` over `send` where possible, even for return types of `caf::result<void>`.

Reply:
I don't follow your argument here. Why should we prefer `request` over `send` if the latter suffices?

Reply:
Because with `send` we silently ignore possibly returned errors from a `caf::result<void>`.

Reply:
So we should use `request().then(...)`. That brings us to the question of what to do in case of an `error`. Sure, we can log it, but the error would come from CAF and we're already in a broken state if we can't communicate with the meta index from here.

Reply:
Exactly. We can log the correct error and propagate it outwards. The necessary measure might be a shutdown.